services/rest/handler.go

```diff
@@ -127,16 +127,60 @@ func (h *Handler) RESTAPIHandler(rw http.ResponseWriter, r *http.Request) {
 	metrics.Increment("batches_read_total", fmt.Sprintf("status=success,conn_group=%s", identifier.Group))
 	h.sendEventCounters(req.Events, identifier.Group)
 
+	resChannel := make(chan struct{}, 1)
 	h.collector.Collect(r.Context(), &collection.CollectRequest{
 		ConnectionIdentifier: identifier,
 		TimeConsumed:         timeConsumed,
 		SendEventRequest:     req,
+		AckFunc:              h.Ack(rw, resChannel, s, req.ReqGuid, identifier.Group),
 	})
+	<-resChannel
+}
 
-	_, err = res.SetCode(pb.Code_CODE_OK).SetStatus(pb.Status_STATUS_SUCCESS).SetSentTime(time.Now().Unix()).
-		SetDataMap(map[string]string{"req_guid": req.ReqGuid}).Write(rw, s)
-	if err != nil {
-		logger.Errorf("[restGetRESTAPIHandler] %s error sending error response: %v", identifier, err)
+func (h *Handler) Ack(rw http.ResponseWriter, resChannel chan struct{}, s serialization.SerializeFunc, reqGuid string, connGroup string) collection.AckFunc {
+	res := &Response{
+		SendEventResponse: &pb.SendEventResponse{},
+	}
+	switch config.Event.Ack {
+	case 0:
```

Comment:
If we agree on the above suggested changes then the `switch` will look more readable

```go
	switch {
	case AckType(Ack) == Receive:
		// do something
		return

	case AckType(Ack) == Publish:
		// do something
		return
	}
```

Reply:
Agreed, will change this